flowpipe/plug.py

```diff
@@ -162,25 +185,52 @@ def __init__(self, name, node, value=None):
             name (str): The name of the Plug.
             node (INode): The Node holding the Plug.
         """
-        super(InputPlug, self).__init__(name, node, (OutputPlug, ))
+        super(InputPlug, self).__init__(name, node, (OutputPlug, SubOutputPlug))
         self.value = value
         self.is_dirty = True
         self.node.inputs[self.name] = self
 
-    def __unicode__(self):
-        """Show this Plug's type and it's connections."""
-        pretty = '\u2192 {0} (IN) {1}'.format(self.name, self.is_dirty)
-        pretty += ''.join(['\n\t\t\u2190 {0}.{1}'.format(
-            c.node.name, c.name) for c in self.connections])
-        return pretty
+    def __getitem__(self, key):
+        """Retrieve a sub plug by key.
+
+        If it does not exist yet, it is created automatically!
+        Args:
+            key (str): The name of the sub plug
+        """
+        if not isinstance(key, str):
+            raise TypeError(
+                'Only strings are allowed as sub-plug keys! '
+                'This is due to the fact that JSON serialization only allows '
+                'strings as keys.')
+        if not self._sub_plugs.get(key):
+            self._sub_plugs[key] = SubInputPlug(
+                name='{0}.{1}'.format(self.name, key),
+                node=self.node,
+                parent_plug=self)
+        return self._sub_plugs[key]
+
+    @property
+    def value(self):
+        """Access to the value on this Plug."""
+        if self._sub_plugs:
+            return {name: plug.value for name, plug in self._sub_plugs.items()}
```

Comment:
Question:
Is it indeed intended that a Plug with SubPlugs can not have a value of its own?

Reply:
Yep, that is the intention, do you see a case where that could be necessary?

Reply:
Not off the top of my head, I just stumbled upon it when reading the code and wanted to make sure it's a conscious design decision.